File: collections/TestRunner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SimpleFramework.Xml {
   public class TestRunner {
      public static void Main(String[] list) {
         TestSuite suite = new TestSuite();
         Type[] types = suite.Suite();
         foreach (Type type in types) {
            Object value = Activator.CreateInstance(type);
            TestCase test = value as TestCase;
            if (test != null) {
               test.Run();
            }
         }
      }
   }
}

File: trunk/download/stream/android/xml/src/org/simpleframework/xml/TestRunner.java
package org.simpleframework.xml;

import java.io.PrintWriter;
import java.io.StringWriter;
import java.util.ArrayList;
import java.util.List;

import junit.framework.TestCase;

public class TestRunner  {  
   private final String[] suites;
   public TestRunner(String... suites) {
      this.suites = suites;
   }
   public String test() throws Exception  {
      List<TestSuite> list = new ArrayList<TestSuite>();
      StringWriter builder = new StringWriter();
      PrintWriter writer = new PrintWriter(builder);
      TestScanner scanner = new TestScanner();
      for(String suite : suites) {
         Class type = Class.forName(suite);
         TestSuite instance = (TestSuite)type.newInstance();
         list.add(instance);
      }
      for(TestSuite suite : list) {
         Class[] tests = suite.suite();
         for(Class type : tests) {
            try {
               TestCase test = (TestCase)type.newInstance();
               scanner.execute(test);
               writer.printf("SUCCESS: %s%n", type);
            }catch(Exception cause) {
               writer.printf("FAILED: %s%n", type);
               cause.printStackTrace(writer);
               return builder.toString();
            }
         }
      }
      return builder.toString();
   }
}